### PR TITLE
Added the Github api fetch functions

### PR DESCRIPTION
**Pull request by Naboth-tieng**
I have implemented the github api so that projects, profile photos, and actions like pull requests are collected. All that is left to do is implement login and signup and improve the CSS.

### CODE REVIEW

**Review by Nisarg4903 — COMMENTED** [review]
solved the conflict

**Comment by Nisarg4903** [comment]
conflict solved, now only a few design changes are left

**Review by Nisarg4903 — COMMENTED** [review]
solved conflict focus on design

File: client/src/components/GithubActivity.jsx
Comment (bedbugs-bit) [inline client/src/components/GithubActivity.jsx:1]: I like the use of 'useState' to manage the `generateGithubActivitiesUrl`

**Review by bedbugs-bit — APPROVED** [review]
Nice work lad!

**Pull request merged**